src/release/breaking-changes/material-state.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Rename MaterialState to WidgetState
+description: >-
+  MaterialState and it's related APIs have been moved
+  outside of the Material library and renamed to
+  WidgetState.
+---
+
+## Summary
+
+`MaterialState`, and its related APIs, have been moved out
+of the Material library and renamed to `WidgetState`.
+
+## Background
+
+Previously, `MaterialState` provided logic for handling multiple different
+states a widget could have, like "hovered", "focused", and
+"disabled". Because this functionality could be useful outside
+of the Material library, namely for the base Widgets layer and
+Cupertino, it was decided to move it outside of Material. As
+part of the move, and to avoid future confusion, the different
+`MaterialState` classes have been renamed to `WidgetState`. The
+behavior of the two are the same.
+
+| Before    | Now |
+| -------- | ------- |
+| `MaterialState` | `WidgetState` |
+| `MaterialStatePropertyResolver` | `WidgetStatePropertyResolver` |
+| `MaterialStateColor` | `WidgetStateColor` |
+| `MaterialStateMouseCursor` | `WidgetStateColorMouseCursor` |
+| `MaterialStateBorderSide` | `WidgetStateBorderSide` |
+| `MaterialStateOutlinedBorder` | `WidgetStateOutlinedBorder` |
+| `MaterialStateTextStyle` | `WidgetStateTextStyle` |
+| `MaterialStateProperty` | `WidgetStateProperty` |
+| `MaterialStatePropertyAll` | `WidgetStatePropertyAll` |
+| `MaterialStatesController` | `WidgetStatesController` |
+| `MaterialStateMixin` | `WidgetStateMixin` |
+
+The classes `MaterialStateOutlineInputBorder` and
+`MaterialStateUnderlineInputBorder` were left in the
+Material library with no `WidgetState` equivilent, as
+they are specific to Material design.
+
+## Migration guide
+
+A [Flutter fix][] is available to help migrate the `MaterialState`
+classes to `WidgetState`.
+
+To migrate, replace `MaterialState` with `WidgetState`.
+
+Code before migration:
+
+```dart
+MaterialState selected = MaterialState.selected;
+
+final MaterialStateProperty<Color> backgroundColor;
+
+class _MouseCursor extends MaterialStateMouseCursor{
+    const _MouseCursor(this.resolveCallback);
+
+    final MaterialPropertyResolver<MouseCursor?> resolveCallback;
+
+    @override
+    MouseCursor resolve(Set<MaterialState> states) => resolveCallback(states) ?? MouseCursor.uncontrolled;
+}
+
+MaterialStateBorderSide? get side;
```

Comment:
This top-level getter without a body looks a little strange. Is that even legal in dart? Maybe just have it return null?